assets/theme-css/code.scss

```diff
@@ -0,0 +1,44 @@
+// Rules for code blocks, including the copy button.
+
+// Selector from "pst/content/_code.scss".
+div[class*="highlight-"],
+div.highlight,
+div.literal-block-wrapper {
+  // Copy-to-clipboard button.
+  & .copy-button {
+    bottom: 10px;
+    float: right;
+    font-size: 0.875em;
+    left: 10px;
+    position: relative;
+    visibility: hidden;
+    z-index: 1;
+  }
+  &:hover .copy-button {
+    visibility: visible;
+  }
+  & .chroma .copy-button button {
+    // Fix alignment that the .chroma rules affect.
+    padding: 0.5rem;
+  }
```

Comment:
The PR looks good overall; is there a reason we cannot adopt the styling from PDST directly? For bootstrap styling, we have to add it, but theirs should already be available to us.

Reply:
I spent about 30 minutes trying to make that work.  The source of the CSS used is obfuscated by being in a Sphinx extension.  So I copied the generated CSS and tried to make that work by adjusting the HTML and CSS and JS, and the differences seemed too great to be worth reconciling; to get the same appearance was going to take a lot more fiddling.  But if you want me to spend more time to try to make that work, just let me know.

Reply:
That's fine, I was more wondering whether we could keep it, without adjustment, and simply render the same elements as they do. If that's not feasible, we'll just put this in as-is.

Reply:
I'll quickly take another look and see if there's a simple approach that I overlooked to make it work that way; if not, I'll merge this as-is.  Thanks.

Reply:
Ok, I've pushed another branch that brings in the relevant Sass and CSS files from PST directly and then overrides the necessary properties in a separate file.  This link shows the difference between this pull request's branch and the branch that uses "more PST": https://github.com/alphapapa/scientific-python-hugo-theme/compare/wip/code-copy-button...alphapapa:scientific-python-hugo-theme:wip/code-copy-button-more-pst?expand=1  The rendering in the browser is the same; the only difference is the amount of code, and a small tweak to the JavaScript.

Please let me know which branch you prefer.  Thanks.

Reply:
Thanks @alphapapa. I like that approach very much.

Note that you don't need to manually copy over the PDST css, you can uncomment the import in `pydata-sphinx-theme.scss`.

Reply:
> Thanks @alphapapa. I like that approach very much.

Ok, I'll see if I can tidy it up a bit more before we merge it.

> Note that you don't need to manually copy over the PDST css, you can uncomment the import in `pydata-sphinx-theme.scss`.

AFAICT uncommenting the `@import "./extensions/copybutton";` line didn't do anything because there was no such file to be imported, because the `_copybutton.scss` file had not been copied from the PDST repo.  As well, the PDST has two relevant files: the `_copybutton.scss` file from the `sphinx-copybutton` extension, and its own customizations in the `copybutton.scss` file.  After copying the first file, the `@import` line worked.

The disadvantage of this approach is that we end up copying more CSS files and rules than we need, and we end up overriding many of the CSS properties to fit our HTML; so if the PDST changes these rules in the future, and we import the files again, we'll likely have to update our overrides again to compensate.

But I'll see if I can harmonize our HTML a bit more with PDST to reduce the number of overrides needed.

Reply:
It's worth keeping the overarching goal in mind here: to get rid of all our own styling.
So that, ultimately, we will have *all* of PDST SCSS in our repo, and almost none of our own.
It feels a bit clumsy now, but it fits better with the future goal.

Your goal of harmonizing our HTML a bit more with theirs is spot-on; thanks!

Reply:
> It's worth keeping the overarching goal in mind here: to get rid of all our own styling. So that, ultimately, we will have _all_ of PDST SCSS in our repo, and almost none of our own. It feels a bit clumsy now, but it fits better with the future goal.
> 
> Your goal of harmonizing our HTML a bit more with theirs is spot-on; thanks!

Agreed, of course.  :)  After Jarrod and I compared the CSS in these two branches, we decided to go with the first branch in this PR so we can go ahead and get this merged.  

But I'll make another to-do item to see if I can find a way to finish harmonizing the HTML and remove all local CSS overrides that sit on top of the PDST CSS so its code can be used exclusively and as-is.  It seems that it will require going over with a fine-toothed comb to find out what's causing the button to not display the same way on SPHT as it does on PDST even with (what seems to be) the same HTML and CSS.